src/registration.ts

```diff
@@ -30,26 +43,33 @@ function init() {
     age: number().required("Number is required")
   });
 
-  // Example of validation, you have to create the same shape on change of validation fields
-  registrationValidatoinShema
-    .validate(
-      {
-        username: "wkdow",
-        email: "email@gmail.com",
-        password: "",
-        age: 2
-      },
-      { abortEarly: false }
-    )
-    .then(validData => {
-      console.log("--handle valid data", validData);
-      formErrorsEl.innerHTML = "";
-    })
-    .catch((validationError: ValidationError) => {
-      console.log("--errors", validationError);
-      const erorrsList = validationError.errors.map(
-        error => `<div class="alert alert-danger" role="alert">${error}</div>`
-      );
-      formErrorsEl.innerHTML = erorrsList.join("");
-    });
+  const onFormChange = (data) => {
+    formDataEl.innerHTML = JSON.stringify(data)
+    registrationValidatoinShema
+      .validate(data, { abortEarly: false })
+      .then(validData => {
+        console.log("--handle valid data", validData);
+        formErrorsEl.innerHTML = "";
+        isValidFormEl.innerHTML = "true"
+      })
+      .catch((validationError: ValidationError) => {
+        console.log("--errors", validationError);
+        const erorrsList = validationError.errors.map(
+          error => `<div class="alert alert-danger" role="alert">${error}</div>`
+        );
+        formErrorsEl.innerHTML = erorrsList.join("");
+        isValidFormEl.innerHTML = "false"
+      });
+  }
+
+  const observables = createObservables(registrationForm)
+
+  combineLatest(observables)
+    .pipe(map(([username, password, email, age]) => ({
```

Comment:
I would better rewrite this line to make it more reusable, like so

`map(errorList => errorList.reduce((acc, val) => ({...acc, ...val}), {}))`

Reply:
dome